codegen/codegen_test.go

```diff
@@ -210,6 +211,29 @@ func TestCodeGen(t *testing.T) {
 				Expect(t, llb.Image("busybox:stable")),
 			)
 		},
+	}, {
+		"parallel coercing fs to group",
+		[]string{"default"},
+		`
+		group default() {
+			parallel group {
+				image "alpine"
+			} fs {
```

Comment:
I think this should be a syntax error, I don't think `fs` builtins should be allowed within a group literal.  If they are allowed I would expect this to work:
```
parallel group {
  image "alpine"
  run "touch foo"
}
```
which won't work due to `run` starting with `scratch` state.  I think the only thing allowed in a group literal is `parallel`, `fs` literals and maybe more `group` literals.

This should work though:
```
parallel fs {
  image "alpine"
} fs {
  ...
}
```

Reply:
We can update this in a subsequent PR

Reply:
I can do this in the same PR. Just a moment.

Reply:
To be clear, I will be adding:
1. Return checker error when `fs` builtin / functions are called within a group block.

And I will not be adding
1. Allowing `fs` literals to be defined in a `group` block. This one will be tougher to contribute.

Reply:
yeah, that makes sense.